director/jobs/models.py

```diff
@@ -0,0 +1,433 @@
+
+    account = models.ForeignKey(
+        Account,
+        on_delete=models.CASCADE,
+        related_name="zones",
+        help_text="The account that this zone is linked to.",
+    )
+
+    name = models.CharField(
+        max_length=256,
+        validators=[
+            validators.RegexValidator(
+                r"^[a-z][a-z0-9\-]*$",
+                "Name should start with a lowercase letter and only contain lowercase letters, digits and hyphens",
+            )
+        ],
+        help_text="The identifier of the queue the job was posted to.",
+    )
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(fields=["account", "name"], name="unique_name")
+        ]
+
+
+class Worker(models.Model):
+    """
+    A worker that runs jobs within a zone.
+
+    This model stores information on a worker as captured by the `overseer` service
+    from Celery's worker monitoring events.
+    """
+
+    zone = models.ForeignKey(
+        Zone,
+        null=True,
+        blank=True,
+        on_delete=models.CASCADE,
+        related_name="workers",
+        help_text="The zone that this worker is in.",
+    )
+
+    created = models.DateTimeField(
+        auto_now_add=True,
+        help_text="The time that the worker started (time of the first event for the worker).",
+    )
+
+    started = models.DateTimeField(
+        null=True,
+        blank=True,
+        help_text="The time that the worker started (only recorded on a 'worker-online' event).",
+    )
+
+    updated = models.DateTimeField(
+        null=True,
+        blank=True,
+        help_text="The time that the last heatbeat was received for the worker.",
+    )
+
+    finished = models.DateTimeField(
+        null=True,
+        blank=True,
+        help_text="The time that the worker finished (only recorded on a 'worker-offline' event)",
+    )
+
+    hostname = models.CharField(
+        max_length=512, help_text="The `hostname` of the worker.",
+    )
+
+    utcoffset = models.IntegerField(help_text="The `utcoffset` of the worker.",)
+
+    pid = models.IntegerField(help_text="The `pid` of the worker.",)
+
+    freq = models.FloatField(help_text="The worker's heatbeat frequency (in seconds)",)
+
+    software = models.CharField(
+        max_length=256, help_text="The name and version of the worker's software.",
+    )
+
+    os = models.CharField(
+        max_length=64, help_text="Operating system that the worker is running on.",
+    )
+
+    signature = models.CharField(
+        max_length=512,
+        help_text="The signature of the worker used to identify it. "
+        "It is possible, but unlikely, that two or more active workers have the same signature.",
+    )
+
+    # The number of missing heartbeats before a worker is considered
+    # as being inactive
+    FLATLINE_HEARTBEATS = 5
+
+    @property
+    def active(self):
+        return (
+            not self.finished
+            and (timezone.now() - self.updated).minutes
```

Comment:
This should be `(timezone.now() - self.updated).seconds`?

Maybe it would be simpler to just have a dead time property e.g.:
`active = (timezone.now() - self.updated).seconds` < self.dead_time`

That seems easier to understand than num heartbeats * heartbeat frequency.

Reply:
I would also suggest splitting this into multiple properties if you do keep the `FLATLINE_HEARTBEATS`. For example, calculate `self.freq * Worker.FLATLINE_HEARTBEATS` in a `max_dead_time_seconds` property. Then do 
`(timezone.now() - self.updated).seconds < self.max_dead_time_seconds`